app/models/agents/data_output_agent.rb

```diff
@@ -169,70 +168,57 @@ def receive_web_request(params, method, format)
 
           return [content, 200]
         else
-          content = Utils.unindent(<<-XML)
-            <?xml version="1.0" encoding="UTF-8" ?>
-            <rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
-            <channel>
-             <atom:link href=#{feed_url(secret: params['secret'], format: :xml).encode(xml: :attr)} rel="self" type="application/rss+xml" />
-             <atom:icon>#{feed_icon.encode(xml: :text)}</atom:icon>
-             <title>#{feed_title.encode(xml: :text)}</title>
-             <description>#{feed_description.encode(xml: :text)}</description>
-             <link>#{feed_link.encode(xml: :text)}</link>
-             <lastBuildDate>#{Time.now.rfc2822.to_s.encode(xml: :text)}</lastBuildDate>
-             <pubDate>#{Time.now.rfc2822.to_s.encode(xml: :text)}</pubDate>
-             <ttl>#{feed_ttl}</ttl>
-
-          XML
-
-          content += simplify_item_for_xml(items).to_xml(skip_types: true, root: "items", skip_instruct: true, indent: 1).gsub(/^<\/?items>/, '').strip
-
-          content += Utils.unindent(<<-XML)
-            </channel>
-            </rss>
-          XML
-
-          return [content, 200, 'text/xml']
+          builder = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
+            xml.rss(:version => '2.0', 'xmlns:atom' => 'http://www.w3.org/2005/Atom') do
+              xml.channel do
+                xml['atom'].link(href: feed_url(secret: params['secret'], format: :xml), rel: 'self', type: 'application/rss+xml')
+                xml['atom'].icon feed_icon
+                xml.title feed_title
+                xml.description feed_description
+                xml.link feed_link
+                xml.lastBuildDate Time.now.rfc2822
+                xml.pubDate Time.now.rfc2822
+                xml.ttl feed_ttl
+                items.each do |item|
+                  xml.item do
+                    build_item_collection_for_xml(item, xml)
+                  end
+                end
+              end
+            end
+          end
+          return [builder.to_xml, 200, 'text/xml']
         end
       end
     end
 
-    private
-
-    class XMLNode
-      def initialize(tag_name, attributes, contents)
-        @tag_name, @attributes, @contents = tag_name, attributes, contents
+    def build_item_collection_for_xml(data, parent = false)
+      return if data.to_s.empty?
+      unless data.is_a?(Hash)
+        parent.text(data)
+        return
       end
 
-      def to_xml(options)
-        if @contents.is_a?(Hash)
-          options[:builder].tag! @tag_name, @attributes do
-            @contents.each { |key, value| ActiveSupport::XmlMini.to_tag(key, value, options.merge(skip_instruct: true)) }
+      data.each do |k, v|
+        if v.is_a?(Hash)
+          attrs = v.fetch('_attributes', {})
+          contents = v.fetch('_contents', v)
+          v.delete('_attributes')
+          v.delete('_contents')
+          parent.send(k, attrs) do
```

Comment:
This might not be safe.  I think `k` comes from a user (Event) and is being sent to an object.